src/components/UIInventory.cpp

```diff
@@ -0,0 +1,219 @@
+#include "UIInventory.hpp"
+#include <GUI/BsGUIButton.h>
+#include <GUI/BsGUILabel.h>
+#include <GUI/BsGUILayoutY.h>
+#include <GUI/BsGUIPanel.h>
+#include <GUI/BsGUIScrollArea.h>
+#include <RTTI/RTTI_UIInventory.hpp>
+#include <components/Inventory.hpp>
+#include <exception/Throw.hpp>
+#include <log/logging.hpp>
+
+namespace REGoth
+{
+  UIInventory::UIInventory(const bs::HSceneObject& parent, HUIElement parentUiElement)
+      : UIElement(parent, parentUiElement, new bs::GUIPanel())
+  {
+    setName("UIInventory");
+
+    mScrollArea = layout().addNewElement<bs::GUIScrollArea>(bs::ScrollBarType::ShowIfDoesntFit,
+                                                            bs::ScrollBarType::NeverShow);
+  }
+
+  UIInventory::~UIInventory()
+  {
+  }
+
+  void UIInventory::setViewedInventory(HInventory inventory)
+  {
+    if (mViewedInventory)
+    {
+      // FIXME: There might be other code registering a callback here, but for some reason
+      //        we can only clear the whole thing! This should be handled differently, but
+      //        I don't know how yet.
+      mViewedInventory->OnItemChanged.clear();
```

Comment:
connect() returns an HEvent you can save somewhere and then call disconnect() on it. 
That should only invalidate this specific connection, right?

Reply:
That's exactly what I needed.